Frontend/CO.CDP.OrganisationApp.Tests/Pages/Organisation/OrganisationRegisterSupplierAsBuyerTests.cs

```diff
@@ -0,0 +1,84 @@
+using CO.CDP.Organisation.WebApiClient;
+using CO.CDP.OrganisationApp.Constants;
+using CO.CDP.OrganisationApp.Models;
+using CO.CDP.OrganisationApp.Pages.Organisation;
+using FluentAssertions;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
+
+namespace CO.CDP.OrganisationApp.Tests.Pages.Organisation;
+public class OrganisationRegisterSupplierAsBuyerTests
+{
+    private readonly Mock<IOrganisationClient> organisationClientMock;
+    private readonly Mock<ITempDataService> tempDataServiceMock;
+    private readonly OrganisationRegisterSupplierAsBuyerModel _model;
+    private readonly Guid orgGuid = new("8b1e9502-2dd5-49fa-ad56-26bae0f85b85");
+
+    public OrganisationRegisterSupplierAsBuyerTests()
+    {
+        organisationClientMock = new Mock<IOrganisationClient>();
+        tempDataServiceMock = new Mock<ITempDataService>();
+        _model = new OrganisationRegisterSupplierAsBuyerModel(tempDataServiceMock.Object, organisationClientMock.Object);
+    }
+
+    [Fact]
+    public async Task ShouldRedirectToSupplierToBuyerOrganisationType_WhenOrganisationIsNotABuyer()
+    {
+        organisationClientMock
+            .Setup(o => o.GetOrganisationAsync(It.IsAny<Guid>()))
+            .ReturnsAsync(new CO.CDP.Organisation.WebApiClient.Organisation(null!, null!, null!, null!, orgGuid, null!, "Org name", new[] { PartyRole.Tenderer }));
```

Comment:
Below comments apply to this and other test methods in this PR.

## named arguments

Can we use named arguments when creating `WebApiClient` model objects please? 

For example:

```csharp
new CO.CDP.Organisation.WebApiClient.Organisation(
            additionalIdentifiers: [],
            addresses:[],
            contactPoint: null,
            id: id ?? Guid.NewGuid(),
            identifier: null,
            name: "Test Org",
            roles: [],
            details: new Details(
                approval: null,
                pendingRoles: pendingRoles != null ? pendingRoles : []
            ));
```

This is desired as arguments are sorted alphabetically, and when a new argument is added it might be injected in the middle of the constructor call. It makes it hard to spot where to add the missing argument value. While, if named arguments are used, the new argument value can be added in any place.

## Test factory

Can we extract the construction of this object to a test factory method please? It takes a while to understand what's important in this object and what's not. Parametrise what's relevant in this and other tests.

For example:

```csharp
organisationClientMock
     .Setup(o => o.GetOrganisationAsync(It.IsAny<Guid>()))
     .ReturnsAsync(GivenOrganisationResponse(organisationId: orgId, roles: [PartyRole.Tenderer]));
```

Reply:
@jakzal Thanks Kuba, I will work on this